core/types/rollup_l1_cost.go

```diff
@@ -81,3 +100,73 @@ func L1Cost(rollupDataGas uint64, l1BaseFee, overhead, scalar *big.Int) *big.Int
 	l1Cost = l1Cost.Mul(l1Cost, scalar)
 	return l1Cost.Div(l1Cost, big.NewInt(1_000_000))
 }
+
+// based on https://github.com/Vectorized/solady/blob/5315d937d79b335c668896d7533ac603adac5315/js/solady.js
```

Comment:
I would like this to be able to be differentially tested against the solidity version which means it will need to be exported. Not sure if the impl should live in `op-geth` or `op-node`, i think either is fine